engine.fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/Dstu3FhirQueryGenerator.java

```diff
@@ -0,0 +1,174 @@
+package org.opencds.cqf.cql.engine.fhir.retrieve;
+
+import java.net.URLDecoder;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.hl7.fhir.dstu3.model.CapabilityStatement;
+import org.hl7.fhir.dstu3.model.CodeType;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.DataRequirement;
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.Duration;
+import org.hl7.fhir.dstu3.model.Period;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.Type;
+import org.hl7.fhir.instance.model.api.IBaseConformance;
+import org.hl7.fhir.instance.model.api.ICompositeType;
+import org.opencds.cqf.cql.engine.elm.execution.SubtractEvaluator;
+import org.opencds.cqf.cql.engine.fhir.model.Dstu3FhirModelResolver;
+import org.opencds.cqf.cql.engine.fhir.searchparam.SearchParameterMap;
+import org.opencds.cqf.cql.engine.fhir.searchparam.SearchParameterResolver;
+import org.opencds.cqf.cql.engine.runtime.Code;
+import org.opencds.cqf.cql.engine.runtime.DateTime;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.cql.engine.terminology.TerminologyProvider;
+
+
+public class Dstu3FhirQueryGenerator extends BaseFhirQueryGenerator {
+    public Dstu3FhirQueryGenerator(SearchParameterResolver searchParameterResolver, TerminologyProvider terminologyProvider, Dstu3FhirModelResolver modelResolver) {
```

Comment:
Same issue here. Downstream projects do not use `Dstu3FhirModelResolver` or `R4FhirModelResolver`. They create their own as appropriate. Need to work out the FHIR Version specific functionality some other way.